doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -1005,7 +1005,162 @@ Response code:
             "temperature"
         ]
 
-## Standard OMA-NGSI operations [/v2/queryContext]
+## Context Subscriptions [/v2/entities/{entityId}/subscriptions]
+
+This section is still under discussion.
+
+A subscription is represented by a JSON object with the following fields:
+
+* 'id': Subscription unique identifier
+* 'subject' : It s an object that describes the subject of the subscription.
+* 'notification' : It is an object that describes the notification received by the subscriber.
+* 'options' : It is a dictionary which can contain different parameters related to the subscription, namely the duration.
+
+A 'subject' contains the following subfields:
+
+* 'entity' : Id or pattern id which identifies the affected entities (optional)
+* 'type' : Type of the affected entities (optional)
+* 'attributes': Attributes to be monitored (if not specified it would mean any)
+* 'condition': Condition that will trigger the notification. It could be 'ONCHANGE' or 'ONTIMEINTERVAL'.
+By default will be ONTIMEINTERVAL. Please note that at the time of writing the editors are considering to combine
+attributes and condition to allow expressions like 'temperature > 25.5'. So this is likely to change.
+
+Entity or type must be present.
+
+A 'notification' object contains the following subfields:
+
+* 'attributes': Attributes to be included in the notification message. If not specified they will be the same as those which are monitored
```

Comment:
Note that the notification will depend on the kind of subscription. The one you describe here is valid for ONCHANGE. However, ONTIMEINTERVAL use a different parameter (a time) and ONVALUE use a condition.

There are two approaches:

* Having separate fields, so notification includes only the field that is relevant for the particular condition. For example, `attributes` for ONCHANGE (as it is now in the text), `interval` for ONTIMEINTERVAL and `expression` for ONVALUE.

* Having a global field (e.g. `condValue`) which value depends on the particular subscription type. This is the approach used by standard subscription.

I think I prefer the former.